Assets/MixedRealityToolkit/Utilities/Editor/Setup/MixedRealityOptimizeWindow.cs

```diff
@@ -0,0 +1,606 @@
+            TotalInactivePolyCountStr = "Total PolyCount (Inactive): "+ totalInActivePolyCount.ToString("N0") + " ";
+
+            var sortedMeshList = meshes.OrderByDescending(s => s.polycount).ToList();
+            for(int i = 0; i < TopListSize; i++)
+            {
+                this.LargestMeshSizes[i] = 0;
+                this.LargestMeshes[i] = null;
+                if (i < meshes.Count)
+                {
+                    this.LargestMeshSizes[i] = sortedMeshList[i].polycount;
+                    this.LargestMeshes[i] = sortedMeshList[i].filter.gameObject;
+                }
+            }
+        }
+
+        private void OptimizeProject()
+        {
+            if (singlePassInstanced)
+            {
+                PlayerSettings.stereoRenderingPath = StereoRenderingPath.Instancing;
+            }
+
+            if (enableDepthBufferSharing)
+            {
+                MixedRealityOptimizeUtils.SetDepthBufferSharing(enableDepthBufferSharing);
+                // TODO: This value needs to be per-perf target
+                //MixedRealityOptimizeUtils.SetDepthBufferFormat(enable16BitDepthBuffer);
+            }
+        }
+
+        private void OptimizeScene()
+        {
+            var lightmapSettings = MixedRealityOptimizeUtils.GetLighmapSettings();
+
+            if (disableRealtimeGlobalIllumination)
+            {
+                MixedRealityOptimizeUtils.ChangeProperty(lightmapSettings, "m_GISettings.m_EnableRealtimeLightmaps", property => property.boolValue = false);
+            }
+
+            if (disableBakedGlobalIllumination)
+            {
+                MixedRealityOptimizeUtils.ChangeProperty(lightmapSettings, "m_GISettings.m_EnableBakedLightmaps", property => property.boolValue = false);
+            }
+        }
+
+        private void DiscoverMaterials()
+        {
+            discoveredMaterials.Clear();
+
+            string[] guids = AssetDatabase.FindAssets("t:Material");
+            for (int i = 0; i < guids.Length; i++)
+            {
+                string assetPath = AssetDatabase.GUIDToAssetPath(guids[i]);
+                Material asset = AssetDatabase.LoadAssetAtPath<Material>(assetPath);
+
+                if (CanConvertMaterial(asset))
+                {
+                    discoveredMaterials.Add(asset);
+                }
+            }
+        }
+        private void ConvertMaterials()
+        {
+            Undo.RecordObjects(this.discoveredMaterials.ToArray(), "Convert to MRTK Standard shader");
+
+            foreach (Material asset in this.discoveredMaterials)
+            {
+                ConvertMaterial(asset);
+            }
+
+            discoveredMaterials.Clear();
+        }
+
+        private void ConvertMaterial(Material asset)
+        {
+            if (asset != null && CanConvertMaterial(asset))
+            {
+                asset.shader = replacementShader;
+            }
+        }
+
+        private bool CanConvertMaterial(Material asset)
```

Comment:
It was eye opening to see how many materials in the MRTK are still using Unity's Standard shader, this tool is awesome! Sadly some of the materials found were imbedded in others assets (often FBXs) and cannot be altered (Unity makes it looks like you can alter them, be when you re-open the editor your changes are lost). Maybe we should filter out materials whose asset path does not end in ".mat"? Or, we could flag them somehow and say "Hey! Extract these materials and please change their shader."

Reply:
>It was eye opening to see how many materials in the MRTK are still using Unity's Standard shader

Right? I couldn't believe it either.